### PR TITLE
feat: add lifecycle script execution

### PR DESCRIPTION
**Pull request by mcataford**
## Description

This PR adds lifecycle script execution to the `next/v2` branch. It mimics what `@yarnpkg/core` does with `scriptUtils.maybeExecuteWorkspaceLifecycleScript`, but in a much simpler way.

Logging was left alone, although there is a path to leverage `StreamReport` if we would like to.

### CODE REVIEW

File: src/utils/maybeExecuteLifecycleScript.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { Workspace, structUtils } from '@yarnpkg/core'
+import { execute } from '@yarnpkg/shell'
+
+import logger from '../logging'
+
+export default async function maybeExecuteLifecycleScript(
+    workspace: Workspace,
+    scriptIdentifier: string,
```
Comment (noahnu) [inline src/utils/maybeExecuteLifecycleScript.ts:8]: (nb)
type LifecycleScriptName = 'prepare' | 'prepack' | 'postpack' // etc...

```suggestion
    scriptIdentifier: LifecycleScriptName,
```

**Comment by mcataford** [comment]
Agreed, will push up a change in that direction. The initial typing was more to broadly validate and get the functionality end to end.

**Review by mcataford — COMMENTED** [review]
Read through, LGTM

**Comment by noahnu** [comment]
LGTM

**Pull request merged**